### ISSUE TITLE
Ledger Nano X Signing Error Bad Argument

### ISSUE THREAD

**Issue by 03dom** [sparrowwallet/sparrow#376]
I have been trying many times to sign a transaction with a ledger nano x from native segwit address to a taproot address and get this failure. Any idea why this could be?

<img width="224" alt="Screenshot 2021-12-30 at 17 47 13" src="https://user-images.githubusercontent.com/35029091/147772663-aee7aa17-2a5b-44fa-bf5e-d047909a8a6d.png">

**Comment by 03dom** [comment]
This may be related
<img width="497" alt="Screenshot 2021-12-30 at 20 33 12" src="https://user-images.githubusercontent.com/35029091/147788022-96ec6343-f9b3-43b5-bfdb-9de1aaf2aeab.png">

**Comment by craigraw** [comment]
This is (probably) because the library Sparrow uses for communicating with the device does not yet support Taproot. However, there has been work done recently to add this, so I expect it to be supported soon.

**Comment by 03dom** [comment]
Ok, thanks. So I will wait for the next release. Awesome wallet by the way, clear, sleek, simple, informative. Great work!

**Comment by 03dom** [comment]
Upgraded to latest version 1.5.6 and problem persists. Cannot sign transaction from native segwit to taproot. Bad Argument (same as above).

**Comment by craigraw** [comment]
Still waiting on the next HWI release. You can track it here: https://github.com/bitcoin-core/HWI

**Comment by orenyomtov** [comment]
Hey @craigraw, was the issue resolved on HWI's end since 2022?

**Comment by craigraw** [comment]
Yes, but for Taproot support on Ledger you will need the Bitcoin app v2.1.0 or later.
